aFWall/src/main/java/dev/ukanth/ufirewall/util/LogNetUtil.java

```diff
@@ -75,24 +81,34 @@ protected String doInBackground(NetParam... params)
 
                         try{
 
-                            /* [WIP] looking for ping timeout option */
-                            String result = output(Runtime.getRuntime().exec(String.format(PING_CMD, /*G.logPingTimeout(), */params[0].address)));
+                            // using libsuperuser
+                            // This will need permission in AFWall+
+                            // "0:(root) Apps running as root"
+
+                            String command = String.format(PING_CMD, Api.getBusyBoxPath(context, true), G.logPingTimeout(), params[0].address);
+
+                            Log.d(TAG, "Execute CMD: " + command);
 
-                            if(result.isEmpty()){
+                            String shell_result = parse(Shell.run("su", new String[]{
```

Comment:
I would do this way

1. if ping present at system level and ping using normal shell
2. else use busybox + ping using su shell 

Hope it makes sense ?